src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+
+    private int size = 0;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+
+    private Node<K, V>[] table = (Node<K,V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+
+    static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putVal(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
+            return table[0].value;
+        }
+        Node<K, V> currentNode = table[hash(key) % table.length];
+        if (currentNode == null) {
+            return null;
+        } else {
+            while (! (currentNode.key != null && currentNode.key.equals(key))) {
+                currentNode = currentNode.next;
+            }
+        }
+        return currentNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static int hash(Object key) {
```

Comment:
all private methods should be after public 
```suggestion
    private int hash(Object key) {
```